res/controllers/Arturia-KeyLab-Mk1-scripts.js

```diff
@@ -0,0 +1,759 @@
+            });
+            this.quickEffectPot = new components.Pot({
+                group: "[QuickEffectRack1_[Channel1]]",
+                midi: [0xB0 + midiChannel, 0x5D],
+                key: "super1",
+                softTakeover: softTakeover,
+            });
+
+            // Other encoders
+            this.valueEncoder = new components.Encoder({
+                group: "[Channel1]",
+                midi: [0xB0 + midiChannel, 0x72], // Value encoder
+                input: function(_channel, _control, value, _status, _group) {
+                    if (!this.inKey) {
+                        // If we haven't selected a parameter yet, don't do
+                        // anything.
+                        return;
+                    }
+                    const param = this.inGetParameter();
+                    if (value === 0x3F) {
+                        if (this.relative) {
+                            this.inSetParameter(-1);
+                            return;
+                        }
+                        const newParam = param - 1;
+                        if (
+                            this.min !== undefined && this.max !== undefined
+                            && newParam < this.min
+                        ) {
+                            this.inSetParameter(this.max);
+                        } else {
+                            this.inSetParameter(param - 1);
+                        }
+                    } else if (value === 0x41) {
+                        if (this.relative) {
+                            this.inSetParameter(1);
+                            return;
+                        }
+                        const newParam = param + 1;
+                        if (
+                            this.max !== undefined && this.min !== undefined
+                            && newParam > this.max
+                        ) {
+                            this.inSetParameter(this.min);
+                        } else {
+                            this.inSetParameter(param + 1);
+                        }
+                    }
+                    this.output(this.inGetParameter());
+                },
+                output: function(value) {
+                    if (this.outputValue) {
+                        const newVal = this.outputValue(value);
+                        writeDisplay(`${this.name}: ${newVal}`);
+                    }
+                },
+            });
+
+            this.valueButton = new components.Button({
+                group: "[Channel1]",
+                midi: [0xB0, 0x73], // Value encoder push
+                encoder: this.valueEncoder,
+                activeDeck: this,
+                inGetValue: function() {
+                    if (!this.inKey) {
+                        return 0;
+                    }
+                    return engine.getValue(this.group, this.inKey.replace("[Channel1]", this.activeDeck.currentDeck));
```

Comment:
why this workaround?

Reply:
Because the key is being set from the menu system it's not getting correctly changed when we swap the deck over. This workaround makes sure it gets changed even if we change the value of inKey after the deck has been selected.